ios/RCTWebRTC/WebRTCModule+RTCMediaStream.m

```diff
@@ -268,12 +268,87 @@ - (void)getUserVideo:(NSDictionary *)constraints
   }
 
   if (videoDevice) {
-    // TODO: Actually use constraints...
-    RTCAVFoundationVideoSource *videoSource = [self.peerConnectionFactory avFoundationVideoSourceWithConstraints:[self defaultMediaStreamConstraints]];
-    // FIXME The effort above to find a videoDevice value which satisfies the
-    // specified constraints was pretty much wasted. Salvage facingMode for
-    // starters because it is kind of a common and hence important feature on
-    // a mobile device.
+     NSMutableDictionary *mandatory = [[NSMutableDictionary alloc]init];
+       NSDictionary *video = constraints[@"video"];
+       RTCMediaConstraints* customConstraints = nil;
+       if (video != nil){
+         NSDictionary *mandatoryParameters = video[@"mandatory"];
+         if(mandatoryParameters != nil){
+           if ([mandatoryParameters valueForKey:@"maxFrameRate"] != nil) {
+             NSMutableString *maxFrameRate = nil;
+             if ([[mandatoryParameters valueForKey:@"maxFrameRate"] isKindOfClass:[NSNumber class]]) {
+               maxFrameRate = [NSMutableString stringWithFormat:@"%@", (NSInteger)mandatoryParameters[@"maxFrameRate"]];
+             }else{
+               maxFrameRate = mandatoryParameters[@"maxFrameRate"];
+             }
+             if (maxFrameRate != nil){
+               [mandatory setObject:maxFrameRate forKey:@"maxFrameRate"];
+             }
+           }
+           if ([mandatoryParameters valueForKey:@"maxHeight"] != nil) {
+             NSMutableString *maxHeight = nil;
+             if ([[mandatoryParameters valueForKey:@"maxHeight"] isKindOfClass:[NSNumber class]]) {
```

Comment:
my opinion for type checking, is to do it on `js` side.
we can check or `toString`, do validation in javascript side
so on the native, we can just set constraint without validation hell.

it's ok to me if you didn't do it in js in the PR.

I can do it as a follow up. just to share my thoughts and my understanding to this repo's general preference.

reduce the line/logic/effort in native code not only make it easier to review/maintain, but also much more agile for adjust / refine / test / experiment. and reduce the size of necessary codes hard-codded in app.

it's the main reason we use `react-native`, isn't it?